samples/boards/st/uart/single_wire/README.rst

```diff
@@ -4,13 +4,16 @@
 
    Use single-wire/half-duplex UART functionality of STM32 devices.
 
-Overview
-********
+# Overview
```

Comment:
Please keep rst formatting.
See https://builds.zephyrproject.io/zephyr/pr/82946/docs/samples/boards/st/uart/single_wire/README.html#uart-stm32-single-wire for the rendering of this change.